src/ripple/app/misc/HashRouter.h

```diff
@@ -149,19 +141,31 @@ class HashRouter
 
     int getFlags (uint256 const& key);
 
-    bool swapSet (uint256 const& key, std::set<PeerShortID>& peers, int flag);
+    /** Prepare the hash entry to be relayed.
+
+        Effects:
+
+            Swaps `peers` with the set of peers that have already
+            been relayed so they can be skipped.
+
+        @return `true` if the item can be relayed. `false` if not.
+           e.g. Because it has been relayed too recently already.
+    */
+    bool prepRelay (uint256 const& key, std::set<PeerShortID>& peers);
```

Comment:
Why swap? In every call `peers` is the empty set. I'm guessing this is done for efficiency. Moving the `peers_` member into the `peers` parameter should be equally efficient and makes for a cleaner interface. (May be outside the scope of this PR.)